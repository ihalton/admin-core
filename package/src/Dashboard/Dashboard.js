import React, { useMemo, useState } from "react";
import { makeStyles } from "@material-ui/core";
import CssBaseline from "@material-ui/core/CssBaseline";
import AppBar from "../AppBar";
import ProfileMenu from "../ProfileMenu";
import NavigationDrawer from "../NavigationDrawer";
import Routes from "../Routes";
import UIContext from "../context/UIContext";
import useMediaQuery from "../hooks/useMediaQuery";
import useRoutes from "../hooks/useRoutes";

const useStyles = makeStyles((theme) => ({
  container: {
    display: "flex"
  },
  leftSidebarOpen: {
    ...theme.mixins.leadingPaddingWhenPrimaryDrawerIsOpen
  }
}));

/**
 * Main dashboard layout
 * @returns {React.ReactElement} A react element representing the main dashboard
 */
function Dashboard() {
  const classes = useStyles();
  const isMobile = useMediaQuery("mobile");
  const routes = useRoutes();
  const [isDetailDrawerOpen, setDetailDrawerOpen] = useState(false);
  const [isNavigationDrawerOpen, setNavigationDrawerOpen] = useState(false);

  const onToggleNavigationDrawer = () => {
    setNavigationDrawerOpen((prevValue) => !prevValue);
  };

  const onToggleDetailDrawer = () => {
    setDetailDrawerOpen((prevValue) => !prevValue);
  };

  const onCloseDetailDrawer = () => {
    setDetailDrawerOpen(false);
  };

  const onCloseNavigationDrawer = () => {
    setNavigationDrawerOpen(false);
  };

  const contextValue = useMemo(() => ({
    isDetailDrawerOpen,
    isMobile,
    isNavigationDrawerOpen: (isMobile && isNavigationDrawerOpen) || isNavigationDrawerOpen,
    onCloseNavigationDrawer,
    onToggleNavigationDrawer,
    onCloseDetailDrawer,
    onToggleDetailDrawer,
    setDetailDrawerOpen,
    setNavigationDrawerOpen
  }), [isDetailDrawerOpen, isMobile, isNavigationDrawerOpen]);

  return (
    <UIContext.Provider value={contextValue}>
      <CssBaseline />
      <div className={classes.container}>
        <AppBar>
          <ProfileMenu size={40} />
        </AppBar>
        <NavigationDrawer />
        <Routes
          isExactMatch={true}
          routes={routes}
        />
      </div>
    </UIContext.Provider>
  );
}

export default Dashboard;
