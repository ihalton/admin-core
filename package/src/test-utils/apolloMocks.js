import getViewer from "../graphql/queries/viewer.graphql";

export default [
  {
    request: {
      query: getViewer
    },
    result: {
      data: {
        viewer: {
          _id: "1234"
        }
      }
    }
  }
];
