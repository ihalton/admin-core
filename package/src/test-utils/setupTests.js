// // Add async/await features
// import "regenerator-runtime/runtime";

// // add some helpful assertions
import "@testing-library/jest-dom/extend-expect";

// // Polyfill fetch
// import "unfetch/polyfill";
