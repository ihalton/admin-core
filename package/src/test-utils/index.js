
import React from "react";
import PropTypes from "prop-types";
import { render } from "@testing-library/react";
import { ThemeProvider } from "@material-ui/core";
import { MemoryRouter } from "react-router-dom";
import { MockedProvider } from "@apollo/react-testing";
import { defaultTheme } from "@reactioncommerce/catalyst";
import i18n from "i18next";
import { initReactI18next } from "react-i18next";
import apolloMocks from "./apolloMocks";

i18n
  .use(initReactI18next)
  .init({
    resources: {
      en: {
        translation: {}
      }
    },
    lng: "en",
    fallbackLng: "en",

    interpolation: {
      escapeValue: false
    }
  });

/**
 * Component that wraps components with mock providers during testing.
 * @return {Component} - Component wrapped with mock providers
 */
const TestProviders = ({ children }) => (
  <MockedProvider mocks={apolloMocks}>
    <ThemeProvider theme={defaultTheme}>
      <MemoryRouter>
        {children}
      </MemoryRouter>
    </ThemeProvider>
  </MockedProvider>
);

TestProviders.propTypes = {
  /** React Component */
  children: PropTypes.element.isRequired
};

/**
 * Custom test renderer that wraps all components with the appropriate mock providers.
 * @param {Component} component - React component to render.
 * @param {Object} options - Options.
 * @return {Object} - @see {@link https://testing-library.com/docs/react-testing-library/api#render-result|react-testing-library}
 */
const renderWithProviders = (component, options) => (
  render(component, { wrapper: TestProviders, ...options })
);

export * from "@testing-library/react";
export { renderWithProviders as render };
