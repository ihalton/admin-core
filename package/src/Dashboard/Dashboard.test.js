import React from "react";
import { render } from "../test-utils";
import Dashboard from "./Dashboard";

jest.mock("../hooks/useAuth", () => ({
  __esModule: true,
  default: () => ({
    logout: jest.fn(),
    viewer: {
      primaryEmail: "test@reactioncommerce.com"
    }
  })
}));

const mockPlugins = [
  {
    route: "/test-1",
    navTitle: "Test 1",
    // eslint-disable-next-line react/display-name
    MainComponent: () => (<span>Route Test 1</span>)
  }
];

test("should be true", () => {
  const { asFragment } = render((
    <Dashboard
      title="Test App"
      plugins={mockPlugins}
    />
  ));
  expect(asFragment()).toMatchSnapshot();
});
