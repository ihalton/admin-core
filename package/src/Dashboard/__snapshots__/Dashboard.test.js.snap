// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should be true 1`] = `
<DocumentFragment>
  <div
    class="MuiContainer-root MuiContainer-maxWidthLg"
  >
    <div
      class="MuiDrawer-root MuiDrawer-docked"
    >
      <div
        class="MuiPaper-root MuiDrawer-paper MuiDrawer-paperAnchorLeft MuiDrawer-paperAnchorDockedLeft MuiPaper-elevation0"
      >
        <ul
          class="MuiList-root MuiList-padding"
        >
          <div
            aria-disabled="false"
            class="MuiButtonBase-root MuiListItem-root makeStyles-listItem-6 MuiListItem-gutters MuiListItem-button"
            role="button"
            tabindex="0"
          >
            <div
              class="MuiListItemText-root makeStyles-listItemText-7"
            >
              Test 1
            </div>
            <span
              class="MuiTouchRipple-root"
            />
          </div>
          <div
            aria-disabled="false"
            class="MuiButtonBase-root MuiListItem-root makeStyles-listItem-6 MuiListItem-gutters MuiListItem-button"
            role="button"
            tabindex="0"
          >
            <div
              class="MuiListItemText-root makeStyles-listItemText-7"
            >
              Sign Out
            </div>
            <span
              class="MuiTouchRipple-root"
            />
          </div>
        </ul>
      </div>
    </div>
    <div
      style="padding-left: 320px;"
    >
      <h1
        class="MuiTypography-root MuiTypography-h1"
      >
        Test App
      </h1>
      <h2
        class="MuiTypography-root MuiTypography-h2"
      >
        Welcome 
      </h2>
    </div>
  </div>
</DocumentFragment>
`;
